index.js

```diff
@@ -396,6 +410,16 @@ function session(options) {
       });
     }
 
+    // check if session has been or is being destroyed
+    function isBeingDestroyed(req) {
+      return req.sessionID && req.session == null;
```

Comment:
Hmm, if I add that check here, the tests don't pass any longer, but if I don't add the check, then `req.session = null` will destroy the cookie even if the user doesn't want `req.session = null` to destroy the session...